internal/pkg/service/appsproxy/proxy/pagewriter/error.go

```diff
@@ -0,0 +1,117 @@
+package pagewriter
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	svcerrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const ExceptionIDPrefix = "keboola-appsproxy-"
+
+type errorPageData struct {
+	Status      int
+	StatusText  string
+	Messages    []string
+	Details     string
+	ExceptionID string
+}
+
+func (pw *Writer) ProxyErrorHandler(w http.ResponseWriter, req *http.Request, err error) {
+	var dnsError *net.DNSError
+	if errors.As(err, &dnsError) {
+		pw.logger.Info(req.Context(), "app is not running, rendering spinner page")
+		pw.WriteSpinnerPage(w, req)
+	}
+
+	pw.WriteError(w, req, svcerrors.NewBadGatewayError(err).WithUserMessage("Request to application failed."))
+}
+
+func (pw *Writer) WriteError(w http.ResponseWriter, req *http.Request, err error) {
```

Comment:
Overall I would split this function into smaller functions if possible. It is quite long, could be separated by error or by status reported.

Reply:
:arrow_right: let's discuss it outside this PR, please add link to the comment to the [your document](https://keboola.atlassian.net/wiki/spaces/ENGG/pages/3301507091/TMP+Martin+Va+ko+Onboarding+Suggestions).

Reply:
:arrow_right: let's discuss it outside this PR, please add link to the comment to the [your document](https://keboola.atlassian.net/wiki/spaces/ENGG/pages/3301507091/TMP+Martin+Va+ko+Onboarding+Suggestions).